doc/reference/dune-workspace/repository.rst

```diff
@@ -0,0 +1,37 @@
+repository
+==========
+
+.. warning::
+
+   Dune Package Management is not final yet and the configuration options
+   are subject to change.
+
+This stanza defines a new named repository and attaches a source location to it.
+
+.. note::
+
+   Defining a repository does not enable it in project by default. It needs to be
+   enabled in a lock directory using the :doc:`/reference/dune-workspace/lock_dir`
```

Comment:
I would also point the `(repositories ...)` stanza here to clarify which one we are talking about.

Reply:
I don't think I can link all that exactly but also `lock_dir` doesn't have that many options :-)